src/_pytest/config/__init__.py

```diff
@@ -140,6 +140,7 @@ def directory_arg(path, optname):
     "stepwise",
     "warnings",
     "logging",
+    "reports",
```

Comment:
Add it to / test it with https://github.com/pytest-dev/pytest/pull/4967/files#diff-4ef9ee85a548e20a4348fa1c8ec445c0R1195 ?

I think it makes sense to have two (internal) lists here already, one being tested, and the other skipped.
`default_plugins` would then be the sum of those.

Reply:
Agree about having two lists, and while at it, we should also stop running with a clear message if one of the essential plugins is passed to `-p no:`, "this internal plugin is essential for pytest function and cannot be blocked".

I would put those plugins in a list `essential_plugins`, and just make `default_plugins` a sum of the essential plugins with the rest:

```python
essential_plugins = (
    "fixtures",
    "helpconfig",  # Provides -p.
    ...
)

default_plugins = essential_plugins + (    
    "debugging",
    "unittest", 
    ...
)    
```

But I believe this is better done in a separate PR, don't you agree?

Reply:
Sounds good.
Yes, in another PR.

Reply:
Thanks for creating the issue, I should have done it myself. 👍 

Anything else? If not could you approve please? 😁

Reply:
Should be added to the test in https://github.com/pytest-dev/pytest/pull/4967/files#diff-4ef9ee85a548e20a4348fa1c8ec445c0R1195 already I think.

Also I would like to see the code parts covered completely.

Reply:
Is coverage the same as with xdist previously?  Is the code moved only?

Reply:
> Also I would like to see the code parts covered completely.

What do you mean? IIUC that test already covers this, `"reports"` is in the `default_plugins` list already. 🤔 

> Is coverage the same as with xdist previously? Is the code moved only?

Yes, I did not changed the code when I moved it.

I see to improve the newly added code:

* In [`reports.py`](https://codecov.io/gh/pytest-dev/pytest/compare/15d608867dfa0fea5de4d9385dd2da8191ec0b8c...579c20c753c754de82796971f897de66b7d767bb/src/src/_pytest/reports.py) there's some moved code which didn't have tests before, I will add coverage for those.

* [capture.py](https://codecov.io/gh/pytest-dev/pytest/compare/15d608867dfa0fea5de4d9385dd2da8191ec0b8c...579c20c753c754de82796971f897de66b7d767bb/changes) seems to be flaky (#4981).

Reply:
Oh, sure regarding `default_plugins`.. :)

Re coverage, I would go from the diff: https://codecov.io/gh/pytest-dev/pytest/compare/15d608867dfa0fea5de4d9385dd2da8191ec0b8c...579c20c753c754de82796971f897de66b7d767bb/diff#D5-191

Reply:
Improved coverage, let's see how it fares now.